packages/gatsby/src/utils/webpack-utils.js

```diff
@@ -290,13 +290,25 @@ module.exports = async ({
   const rules = {}
 
   /**
-   * JavaScript loader via babel, excludes node_modules
+   * JavaScript loader via babel, includes userland code
+   * and packages that depend on `gatsby`
    */
   {
-    let js = (options = {}) => {
+    let js = (
+      { modulesThatUseGatsby, ...options } = { modulesThatUseGatsby: [] }
+    ) => {
       return {
         test: /\.(js|mjs|jsx)$/,
-        exclude: vendorRegex,
+        include: modulePath => {
+          if (vendorRegex.test(modulePath)) {
```

Comment:
I would write this with an early return, doesn't really matter but that's my code style, no big deal though 😛 
```js
// when it's not coming from node_modules we treat it as a source file.
if (!vendorRegex.test(modulePath)) {
  return true;
}

// If the module uses Gatsby as a dependency we want to treat it as src so we can extract queries, ...
return modulesThatUseGatsby.some(module =>
  modulePath.includes(module.path)
)
```

Reply:
This is much better! Thank you